src/movies.js

```diff
@@ -1,16 +1,85 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(allMovies) {
+    const result = allMovies.map(elm => {
+        return elm.director
+    })
+    return result
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(allMovies) {
+    const result = allMovies.filter(elm => {
+        if (elm.director === 'Steven Spielberg' && elm.genre.includes('Drama')) {
+            return true
+        }
+    })
+    return result.length
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(allMovies) {
+    if (allMovies.length === 0) {
+        return 0
+    }
+    const sumRate = allMovies.reduce((acc, elm) => {
+
+        if (typeof elm.rate != 'number') {
+            return acc
+        }
+        return acc + elm.rate
+
+
+    }, 0)
+    return parseFloat((sumRate / allMovies.length).toFixed(2))
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(allMovies) {
+    const dramaMovies = allMovies.filter(elm => {
+        if (elm.genre.includes('Drama')) {
+            return true
+        }
```

Comment:
includes ya devuelve true o false, por lo que se puede retornar directamente.
```suggestion
        return elm.genre.includes('Drama')
```